extensions/amp-analytics/0.1/events.js

```diff
@@ -1506,32 +1508,37 @@ export class VisibilityTracker extends EventTracker {
       // Array selectors do not suppor the special cases: ':host' & ':root'
       const selectionMethod =
         config['selectionMethod'] || visibilitySpec['selectionMethod'];
-      const selectors = Array.isArray(selector) ? selector : [selector];
-      for (let i = 0; i < selectors.length; i++) {
-        unlistenPromises.push(
-          this.root
-            .getAmpElement(
-              context.parentElement || context,
-              selectors[i],
-              selectionMethod,
-              multiSelectorVisibilityOn
-            )
-            .then(element =>
+      const selectors = Array.isArray(selector)
+        ? selector.filter((val, index) => selectors.indexOf(val) === index)
```

Comment:
I'm curious if this actually works. You are trying to use `selectors` before initialization? 
And I think we should definitely throw a user error if selector contains duplicate entry.

Reply:
Ok, I'll add verification here